cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java

```diff
@@ -228,4 +230,16 @@ public void error(Throwable cause) {
     }
     return controller;
   }
+
+  @Nullable
+  private String getAdapterName(Arguments arguments) {
+    // TODO: Currently this logic is super ugly, should try and pass adapter name as its own separate arg
+    if (arguments.hasOption(ProgramOptionConstants.SCHEDULE_NAME)) {
+      String scheduleName = arguments.getOption(ProgramOptionConstants.SCHEDULE_NAME);
+      if (scheduleName.contains(".")) {
```

Comment:
So this actually follows the following class hierarchy:

*Adapter* -> *CDAP Schedule* -> *Quartz Schedule*

The ``ProgramOption``s are set in a class (``DefaultSchedulerService``) that only gets a Quartz class - ``JobExecutionContext``. And we do not pass any context information from *CDAP Schedule* classes to the Quartz Scheduler classes other than the trigger key which is an encoded string of the form: ``namespaceId:appId:programType:programId:scheduleName``.